x/interchainstaking/keeper/ibc_packet_handlers.go

```diff
@@ -139,22 +130,15 @@ func (k *Keeper) HandleAcknowledgement(ctx sdk.Context, packet channeltypes.Pack
 			}
 			response := lsmstakingtypes.MsgRedeemTokensforSharesResponse{}
 
-			if msgResponseType != "" {
```

Comment:
Each one of these cases was doing the same logic for msgResponse and then msgData.Data - refactored for less code and greater simplicity.